src/components/ConsoleInput.jsx

```diff
@@ -14,12 +15,33 @@ export default class ConsoleInput extends Component {
     bindAll(this, '_ref');
   }
 
-  componentWillReceiveProps({isTextSizeLarge}) {
-    if (isTextSizeLarge !== this.props.isTextSizeLarge) {
+  componentDidMount() {
+    this._focusRequestedLine(this.props.requestedFocusedLine);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isTextSizeLarge !== this.props.isTextSizeLarge) {
       requestAnimationFrame(() => {
         inheritFontStylesFromParentElement(this._editor);
       });
     }
+    this._focusRequestedLine(nextProps.requestedFocusedLine);
+  }
+
+  _focusRequestedLine(requestedFocusedLine) {
+    if (get(requestedFocusedLine, 'language') !== 'console') {
```

Comment:
Let’s generalize the structure of `requestedFocusedLine`—instead of `language` let’s maybe `component`, and just do `editor.html`, `editor.css`, `console`, etc. Reasonable?